mod/async/pkg/types/event.go

```diff
@@ -25,56 +25,72 @@ import (
 	"errors"
 )
 
-// EventID represents the type of an event.
+// EventID represents the type of a message.
 type EventID string
 
-// Event represents a generic event in the beacon chain.
-type Event[DataT any] struct {
-	// ctx is the context associated with the event.
-	ctx context.Context
-	// eventType is the name of the event.
-	eventType EventID
-	// event is the actual beacon event.
-	data DataT
-	// error is the error associated with the event.
-	err error
+// BaseEvent defines the minimal interface that the dispatcher expects from a
+// message.
+type BaseEvent interface {
+	ID() EventID
+	Context() context.Context
+}
+
+// Event defines the interface that the underlying route expects from a
+// message with data.
+type Event[DataT any] interface {
+	BaseEvent
+	Data() DataT
+	Error() error
+	Is(id EventID) bool
 }
 
 // NewEvent creates a new Event with the given context and beacon event.
 func NewEvent[
```

Comment:
imo constructor should be below the type def, also move type to events @archbear you still think thats the right call?

Reply:
oops forgot to delete this file